### ISSUE TITLE
Failed compile

### ISSUE THREAD

**Issue by PikekTeam** [sensepost/objection#49]
Hi, 

I tried to use Objection on HuluPlus just as a test, and it failed. I'm new to this and this method but not new to modifying apps (for Android anyway).

I was hoping you could shed some light on what is happening here. I'll include the full trace. I tried with Groupon just as a test and all went fine.

```
Using Gadget version: 10.6.41
Unpacking hulu.apk
App already has android.permission.INTERNET
Reading smali from: /tmp/tmprwl_cthi.apktemp/smali/com/hulu/features/splash/SplashActivity.smali
Injecting loadLibrary call at line: 22
Writing patched smali back to: /tmp/tmprwl_cthi.apktemp/smali/com/hulu/features/splash/SplashActivity.smali
Creating library path: /tmp/tmprwl_cthi.apktemp/lib/arm64
Copying Frida gadget to libs path...
Rebuilding the APK with the frida-gadget loaded...
Rebuilding the APK may have failed. Read the following output to determine if apktool actually had an error:                                                          
W: /tmp/tmprwl_cthi.apktemp/AndroidManifest.xml:101: Tag <receiver> attribute name has invalid character '}'. Ignoring...                                             
W: /tmp/tmprwl_cthi.apktemp/AndroidManifest.xml:102: Tag <service> attribute name has invalid character ']'. Ignoring...                                              
W: /tmp/tmprwl_cthi.apktemp/AndroidManifest.xml:103: Tag <service> attribute name has invalid character ''. Ignoring...                                               
W: /tmp/tmprwl_cthi.apktemp/AndroidManifest.xml:116: Tag <activity> attribute name has invalid character '*'. Ignoring...                                            
W: /tmp/tmprwl_cthi.apktemp/assets/LICENSE.txt: error: File is case-insensitive equivalent to: /tmp/tmprwl_cthi.apktemp/assets/license.txt                            
Exception in thread "main" brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/tmp/brut_util_Jar_3725730383723196938.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 25, --version-code, 250374, --version-name, 3.16.0.250374, --no-version-vectors, -F, /tmp/tmprwl_cthi.apktemp.objection.apk, -0, arsc, -0, arsc, -A, /tmp/tmprwl_cthi.apktemp/assets, /tmp/tmprwl_cthi.apktemp/build/apk]                                                   at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:456)                                                                                        at brut.androlib.Androlib.buildApk(Androlib.java:689)                              
at brut.androlib.Androlib.build(Androlib.java:309)                                 
at brut.androlib.Androlib.build(Androlib.java:270)                                 
at brut.apktool.Main.cmdBuild(Main.java:227)                                       
at brut.apktool.Main.main(Main.java:75)                                    
Caused by: brut.common.BrutException: could not exec (exit code = 1): [/tmp/brut_util_Jar_3725730383723196938.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 25, --version-code, 250374, --version-name, 3.16.0.250374, --no-version-vectors, -F, /tmp/tmprwl_cthi.apktemp.objection.apk, -0, arsc, -0, arsc, -A, /tmp/tmprwl_cthi.apktemp/assets, /tmp/tmprwl_cthi.apktemp/build/apk]                 
at brut.util.OS.exec(OS.java:95)                                                   
at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:450)                                                                                        ... 5 more                                                                 

Built new APK with injected loadLibrary and frida-gadget
Signing new APK.
jarsigner: unable to open jar file: /tmp/tmprwl_cthi.apktemp.objection.apk

Signed the new APK
Performing zipalign
Zipaligning the APK may have failed. Read the following output to determine if zipalign actually had an error:                                                        Unable to open '/tmp/tmprwl_cthi.apktemp.objection.apk' as zip archive

Zipaling completed
Copying final apk from /tmp/tmprwl_cthi.apktemp.aligned.objection.apk to current directory...
Traceback (most recent call last):
  File "/usr/local/bin/objection", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 1066, in invoke    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/objection/console/cli.py", line 213, in patchapk
    patch_android_apk(**locals())
  File "/usr/local/lib/python3.5/dist-packages/objection/commands/mobile_packages.py", line 156, in patch_android_apk
    shutil.copyfile(patcher.get_patched_apk_path(), os.path.join(os.path.abspath('.'), destination))
  File "/usr/lib/python3.5/shutil.py", line 114, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmprwl_cthi.apktemp.aligned.objection.apk'
Cleaning up temp files...
Failed to cleanup with error: [Errno 2] No such file or directory: '/tmp/tmprwl_cthi.apktemp.objection.apk' 
```

Thanks for any insight and very cool/exciting project you're working on here!

**Comment by PikekTeam** [comment]
(I forgot to ask if this was maybe an issue with multidex, but I suppose you'll know more than I. I didn't manually verify that Groupon wasn't multidex but I assume that Hulu is)

**Comment by leonjza** [comment]
Hey! Thanks for reporting this. I'll test this asap.

**Comment by leonjza** [comment]
I have released v1.2.5, which should hopefully solve this. If there are cases where you might need resources to be decoded, add the new `--decode-resources` flag.

`pip3 install objection --upgrade`, test and let me know how it goes!
